Assets/Scripts/Augment/AugmentImplementations/Hat/HatBarrel.cs

```diff
@@ -136,13 +138,23 @@ private void UpdateProjectile(ProjectileState state)
 
         if (collisions.Length > 0)
         {
-            state.active = false;
-            OnColliderHit?.Invoke(collisions[0], ref state);
             HitboxController hitbox = collisions[0].GetComponent<HitboxController>();
```

Comment:
TryGetComponent instead?

Reply:
That would be better to do while I'm at it, I agree